include/spdlog/sinks/base_sink-inl.h

```diff
@@ -53,7 +53,16 @@ void SPDLOG_INLINE spdlog::sinks::base_sink<Mutex>::set_formatter(std::unique_pt
 template<typename Mutex>
 void SPDLOG_INLINE spdlog::sinks::base_sink<Mutex>::set_pattern_(const std::string &pattern)
 {
-    set_formatter_(details::make_unique<spdlog::pattern_formatter>(pattern));
+    const auto current_pattern_formatter = dynamic_cast<spdlog::pattern_formatter *>(formatter_.get());
+
+    if (current_pattern_formatter)
+    {
+        current_pattern_formatter->set_pattern(pattern);
+    }
+    else
+    {
+        set_formatter_(details::make_unique<spdlog::pattern_formatter>(pattern));
+    }
```

Comment:
@gabime I see you rejected my pull request, unfortunately, but could you possibly still consider this change? When a sink already has a pattern_formatter, why not call `set_pattern(pattern)` on its current formatter, instead of creating a new one?

Reply:
Why is it needed?

Reply:
It preserves the original pattern_formatter properties of the sink, when a new pattern is set. Specifically, it preserves the original pattern_formatter::eol_ string.

Reply:
This dynamic cast can be problematic(I think rtti would be required on some setups). 
You can use  `set_formatter(new_formatter)` instead.

Reply:
> This dynamic cast can be problematic

OK, I understand, on platforms that don't have rtti, `dynamic_cast` could be a problem. Still it seems better to me if `base_sink<Mutex>::set_pattern_` could only just set the pattern string of its `formatter_`, while leaving the other `pattern_formatter` data unchanged. Anyway, that's just my impression.